app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/mapper/SwapWidgetMapper.kt

```diff
@@ -0,0 +1,138 @@
+            )
+        }
+    }
+
+    fun mapTokenA(
+        token: SwapTokenModel,
+        tokenAmount: BigDecimal? = null,
+    ): SwapWidgetModel {
+        return SwapWidgetModel.Content(
+            isStatic = false,
+            widgetTitle = swapWidgetFromTitle(),
+            availableAmount = availableAmount(token),
+            balance = balance(token),
+            currencyName = tokenName(token),
+            amount = tokenAmount(token, tokenAmount),
+            fiatAmount = fiatAmount(token, tokenAmount),
+            amountMaxDecimals = token.decimals,
+        )
+    }
+
+    fun mapTokenB(token: SwapTokenModel, tokenAmount: BigDecimal?): SwapWidgetModel {
+        var fiatAmount = fiatAmount(token, tokenAmount)
+        if (true) {
+            // todo price impact
+            fiatAmount = fiatAmount?.copy(textColor = R.color.text_night)
+        }
+        return SwapWidgetModel.Content(
+            isStatic = true,
+            widgetTitle = swapWidgetToTitle(),
+            availableAmount = null,
+            balance = balance(token),
+            currencyName = tokenName(token),
+            amount = tokenAmount(token, tokenAmount),
+            fiatAmount = fiatAmount,
+            amountMaxDecimals = token.decimals,
+        )
+    }
+
+    fun mapTokenBLoading(
+        token: SwapTokenModel,
+    ): SwapWidgetModel.Content =
+        SwapWidgetModel.Content(
+            isStatic = true,
+            widgetTitle = swapWidgetToTitle(),
+            currencyName = tokenName(token),
+            amount = textCellSkeleton(
+                height = 20.toPx(),
+                width = 84.toPx(),
+                radius = 6f.toPx(),
+            ),
+            balance = balance(token),
+            fiatAmount = null,
+            availableAmount = null,
+        )
+
+    private fun fiatAmount(token: SwapTokenModel, tokenAmount: BigDecimal?): TextViewCellModel.Raw? {
+        if (tokenAmount == null) return null
+        val ratio = when (token) {
+            is SwapTokenModel.JupiterToken -> token.token.priceInUsd
+            is SwapTokenModel.UserToken -> token.token.totalInUsd
+        } ?: return null
+
+        val usd = tokenAmount.multiply(ratio).formatFiat()
+        return TextViewCellModel.Raw(TextContainer(R.string.swap_main_fiat_value, usd))
+    }
+
+    private fun tokenAmount(token: SwapTokenModel, tokenAmount: BigDecimal?): TextViewCellModel.Raw {
+        val decimals = token.decimals
+        val amountText = tokenAmount?.formatToken(decimals) ?: "0"
+        return TextViewCellModel.Raw(TextContainer(amountText))
+    }
+
+    private fun tokenName(token: SwapTokenModel): TextViewCellModel.Raw =
+        TextViewCellModel.Raw(
+            TextContainer(
+                when (token) {
+                    is SwapTokenModel.JupiterToken -> token.token.tokenName
+                    is SwapTokenModel.UserToken -> token.token.tokenName
+                }
```

Comment:
that looks weird. 
Just add one fun in `SwapTokenModel` which checks type and returns `tokenName`